_posts/2019-02-02-digitalocean-one-click.md

```diff
@@ -0,0 +1,193 @@
+
+![Click create](/images/digitalocean-one-click/create-droplet.png)
+
+### Access the Droplet
+
+Once the Droplet is ready it will appear in your list of Droplets within the DigitalOcean resources pane.
+
+![Droplet is ready](/images/digitalocean-one-click/droplet-ready.png)
+
+Copy the IP address and using your favourite terminal log in to the Droplet using `root@<IP address>`
+
+> Note: ensure you have made the appropriate private key available to the client you are using.
+
+Once logged in follow the instructions; at the time of writing this is simply to press Enter.
+
+![Logged in](/images/digitalocean-one-click/terminal.png)
+
+A number of configuration actions will be performed for you, and the result will be a running Swarm-based OpenFaaS instance.  Make a note of the credentials - in particular the final line which will set our credentials in the `faas-cli` - we'll need this when we deploy our function.  We can now close the connection to the Droplet.
+
+* On your local machine set the `OPENFAAS_URL`:
+
+```
+$ export OPENFAAS_URL=http://159.65.92.17:8080
+```
+
+* Run the CLI login command:
+
+```
+echo -n 6acf4e5dc4997530666fece7be3e15e5a1b7ba9572163aa2beb8ff9449ee1911 | faas-cli login --username=admin --password-stdin
+```
+
+These will save us time and effort when we come to deploy our function to the Droplet.
+
+### Create a function
+
+We're going to create a simple Python function that when invoked echoes back a string.
+
+* Create a workspace for our function:
+
+```
+$ mkdir -p ~/openfaas/functions/ && cd ~/openfaas/functions/
+```
+
+* Create a function called `hello-digitalocean` using the CLI:
+
+> Replace the `rgee0` prefix with your own Docker hub account name so as to enable the ensuing `push` action.
+
+```
+$ faas-cli new --lang=python --prefix=rgee0 hello-digitalocean
+
+faas-cli new --lang=python --prefix=rgee0 hello-digitalocean
+2019/02/02 11:42:08 No templates found in current directory.
+2019/02/02 11:42:08 Attempting to expand templates from https://github.com/openfaas/templates.git
+2019/02/02 11:42:10 Fetched 15 template(s) : [csharp csharp-armhf dockerfile go go-armhf java8 node node-arm64 node-armhf php7 python python-armhf python3 python3-armhf ruby] from https://github.com/openfaas/templates.git
+Folder: hello-digitalocean created.
+  ___                   _____           ____
+ / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
+| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
+| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
+ \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
+      |_|
+
+
+Function created in folder: hello-digitalocean
+Stack file written: hello-digitalocean.yml
+```
+
+* Rename `hello-digitalocean.yml` to `stack.yml`:
```

Comment:
Not about the blog, but should we change the CLI to look first for stack.yml then any YAML file found in the directory to stop people having to rename to stack.yml in every blog post / guide?

Reply:
Depends how people lay out their functions.  If they have multiple functions / multiple ymls in the same top level folder then I'm not sure what this behaviour would look like.